processing/threedidepth_algorithm.py

```diff
@@ -0,0 +1,293 @@
+from qgis.core import (
+    QgsFeedback,
+    QgsProcessingAlgorithm,
+    QgsProcessingParameterEnum,
+    QgsProcessingParameterFile,
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterDestination,
+    QgsProcessingParameterRasterLayer,
+)
+
+import h5py
+from threedidepth.calculate import calculate_waterdepth
+from threedidepth.calculate import (
+    MODE_LINEAR,
+    MODE_CONSTANT,
+    MODE_LINEAR_S1,
+    MODE_CONSTANT_S1,
+    MODE_LIZARD,
+    MODE_LIZARD_S1
+)
+
+
+logger = logging.getLogger(__name__)
+pluginPath = os.path.split(os.path.dirname(__file__))[0]
+Mode = namedtuple("Mode", ["name", "description"])
+
+
+class ProcessingParamterNetcdfNumber(QgsProcessingParameterNumber):
+    def __init__(self, *args, parentParameterName='', **kwargs):
+        super().__init__(*args, **kwargs)
+        self.parentParameterName = parentParameterName
+        self.setMetadata(
+            {'widget_wrapper': {'class': ThreediResultTimeSliderWidget}}
+        )
+
+
+WIDGET, BASE = uic.loadUiType(os.path.join(
+    pluginPath, 'processing', 'ui', 'widgetTimeSlider.ui')
+)
+
+
+class TimeSliderWidget(BASE, WIDGET):
+    """Timeslider form widget
+
+    Provide a horizontal slider and an LCD display connected to the slider.
+    """
+
+    def __init__(self):
+        super(TimeSliderWidget, self).__init__(None)
+        self.setupUi(self)
+        self.horizontalSlider.valueChanged.connect(self.set_lcd_value)
+        self.index = None
+        self.timestamps = None
+        self.reset()
+
+    def getValue(self):
+        return self.index
+
+    def set_timestamps(self, timestamps):
+        self.setDisabled(False)
+        self.horizontalSlider.setMinimum(0)
+        self.horizontalSlider.setMaximum(len(timestamps) - 1)
+        self.timestamps = timestamps
+
+    def set_lcd_value(self, index: int):
+        self.index = index
+        if self.timestamps is not None:
+            value = self.timestamps[index]
+        else:
+            value = 0
+        lcd_value = self.format_lcd_value(value)
+        self.lcdNumber.display(lcd_value)
+
+    def format_lcd_value(self, value: float) -> str:
+        days = int(value // 86400)
+        hours = int((value // 3600) % 24)
+        minutes = int((value // 60) % 60)
+        formatted_display = "{:d} {:02d}:{:02d}".format(days, hours, minutes)
+        return formatted_display
+
+    def reset(self):
+        self.setDisabled(True)
+        self.index = None
+        self.timestamps = None
+        self.horizontalSlider.setMinimum(0)
+        self.horizontalSlider.setMaximum(0)
+        self.horizontalSlider.setValue(0)
+
+    def new_file_event(self, file_path):
+        """New file has been selected by the user
+
+        Try to read in the timestamps from the file
+        """
+        try:
+            with h5py.File(file_path, 'r') as results:
+                timestamps = results['time'].value
+                self.set_timestamps(timestamps)
+        except Exception:
```

Comment:
maybe good to let the user know that something went wrong?

Reply:
Yeah good point!